tests/ignite/contrib/metrics/regression/test_canberra_metric.py

```diff
@@ -30,18 +31,27 @@ def test_compute():
 
     m = CanberraMetric()
 
+    canberra = DistanceMetric.get_metric("canberra")
+
     m.update((torch.from_numpy(a), torch.from_numpy(ground_truth)))
-    np_sum = (np.abs(ground_truth - a) / (a + ground_truth)).sum()
+    np_sum = (np.abs(ground_truth - a) / (np.abs(a) + np.abs(ground_truth))).sum()
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([a, ground_truth])[0][1] == pytest.approx(np_sum)
 
     m.update((torch.from_numpy(b), torch.from_numpy(ground_truth)))
-    np_sum += ((np.abs(ground_truth - b)) / (b + ground_truth)).sum()
+    value = ((np.abs(ground_truth - b)) / (np.abs(b) + np.abs(ground_truth))).sum()
+    np_sum += value
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([b, ground_truth])[0][1] == pytest.approx(value)
 
     m.update((torch.from_numpy(c), torch.from_numpy(ground_truth)))
-    np_sum += ((np.abs(ground_truth - c)) / (c + ground_truth)).sum()
+    value = ((np.abs(ground_truth - c)) / (np.abs(c) + np.abs(ground_truth))).sum()
+    np_sum += value
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([c, ground_truth])[0][1] == pytest.approx(value)
 
     m.update((torch.from_numpy(d), torch.from_numpy(ground_truth)))
-    np_sum += (np.abs(ground_truth - d) / (d + ground_truth)).sum()
+    value = (np.abs(ground_truth - d) / (np.abs(d) + np.abs(ground_truth))).sum()
+    np_sum += value
     assert m.compute() == pytest.approx(np_sum)
+    assert canberra.pairwise([d, ground_truth])[0][1] == pytest.approx(value)
```

Comment:
Can you also test the total computed metric with sklearn and not a single update ? Same for above.

Reply:
@vfdev-5 done